### PR TITLE
docs: Update Tutorial

### PR DESCRIPTION
**Pull request by richb-hanover**
This is my final pass over the Tutorial section. Many minor changes to the explanations, following the new structure that @aljazerzen created.

I created an "Annotated Example" page that can explain each of the transforms in the default PRQL query shown in the Playground. It's not linked in, but I plan to come back to this when I have a bit more time.

### CODE REVIEW

**Comment by max-sixty** [comment]
Can we merge `main`? That will fix the test (ref https://github.com/PRQL/prql/pull/3052#issuecomment-1693519072)

File: web/book/src/tutorial/aggregation.md [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
-`average` or `sum` &mdash; that reduces the table to a single row.
+`average` or `sum` &mdash; that reduces values in the table to a single row.
 
 ### `aggregate` transform
 
 The `aggregate` transform takes a tuple to create one or more new columns that
 "distill down" data from all the rows.
 
-```
+```prql no-eval
```
Comment (max-sixty) [inline web/book/src/tutorial/aggregation.md:12]: Why are these all `no-eval`?

File: web/book/src/tutorial/filtering.md [hunk 2]
```diff
@@ -1,46 +1,46 @@
 # Filtering rows
 
-In previous page we learned how `select`, `derive`, and `join` change the
+In the previous page we learned how `select`, `derive`, and `join` change the
 columns of a table.
 
-Now we will look into how we can manipulate rows of a table using `filter` and
+Now we will look into how to manipulate the rows of a table using `filter` and
```
Comment (max-sixty) [inline web/book/src/tutorial/filtering.md]: Good changes, thanks.

Comment (max-sixty) [reply web/book/src/tutorial/filtering.md]: and maybe?
```suggestion
Now we will explore how to manipulate the rows of a table using `filter` and
```

File: web/book/src/tutorial/filtering.md [hunk 3]
```diff
@@ -54,11 +54,11 @@ take 3
...
-compiles these two statements/transforms (`take 11..20` and `take 3`) to SQL, it
-produces the same result as `take 11..13`.
+the first 3 rows from that result. This could have been done in a single
+transform `take 11..13`
+
+However, this is a nice example of how PRQL allows fast data exploration by
+"stacking" transforms in the pipeline. PRQL eliminates the worry about
+interactions of a new transform with the previous query. When PRQL compiles
+these two statements/transforms (`take 11..20` and `take 3`) to SQL, it produces
+the same result as `take 11..13`.
```
Comment (max-sixty) [inline web/book/src/tutorial/filtering.md]:
`````suggestion
the first 3 rows from that result.

```admonish note
Note that this a single
transform `take 11..13` would have produced the same SQL. The example 
serves an example of how PRQL allows fast data exploration by
"stacking" transforms in the pipeline, reducing the cognitive burden of how
a new transform with the previous query.
```

`````

File: web/book/src/tutorial/relations.md [hunk 4]
```diff
@@ -16,19 +16,19 @@ looks like this:
 | 5          | 23          | Boston       |       ...       | 13.86 |
 | 6          | 37          | Frankfurt    |       ...       | 0.99  |
 
-A relation is composed of rows, each of which is a tuple containing the same set
-of columns. Each column has an unique name and a designated data type. The table
-above has `invoice_id`, `customer_id`, and `artist_id` columns with a data type
-of "integer number", a `billing_city` column with a data type of "text", a
-number of other columns, and and a `total` column that contains floating-point
+A relation is composed of rows, each of which is a tuple (a list) that contains
+similar columns. Each column has an unique name and a designated data type. The
```
Comment (max-sixty) [inline web/book/src/tutorial/relations.md]:
```suggestion
A relation is composed of rows. Each row in a relation contains
a value for each of the relation's columns. Each column in a relation has an unique name 
and a designated data type. The
```

I actually don't think we need to mention `tuple` here — we introduce it slightly later

**Review by max-sixty — COMMENTED** [review]
Good improvements, thanks!

File: web/book/src/tutorial/aggregation.md (hunk 1 above)
Comment (richb-hanover) [reply web/book/src/tutorial/aggregation.md:12]: My understanding the `prql` gives the syntax coloring; the `no-eval` doesn't create the corresponding SQL block.

Comment (max-sixty) [reply web/book/src/tutorial/aggregation.md:12]: Yes re adding `prql`!

Do we want the SQL block though? Or this is intended for folks who don't know SQL? (Though in that case we are going to have to make it more basic I think? We're talking about relations & tuples, we might need to analogize to a spreadsheet...)

Thoughts @aljazerzen ?

File: web/book/src/tutorial/filtering.md [hunk 5]
```diff
@@ -54,11 +54,12 @@ take 3
...
-data exploration.
-
-We have the freedom to stack transforms on top of each other, without worrying
-about interactions of a new transform with the previous query. When PRQL
-compiles these two statements/transforms (`take 11..20` and `take 3`) to SQL, it
-produces the same result as `take 11..13`.
+the first 3 rows from that result.
+
+```admonish note
+Note that this a single
```
Comment (max-sixty) [inline web/book/src/tutorial/filtering.md]:
```suggestion
Note that a single
```

Comment (max-sixty) [reply web/book/src/tutorial/filtering.md]: (my mistake I think)

File: web/book/src/tutorial/relations.md [hunk 6]
```diff
@@ -16,19 +16,19 @@ looks like this:
...
 
-A relation is composed of rows, each of which is a tuple containing the same set
-of columns. Each column has an unique name and a designated data type. The table
-above has `invoice_id`, `customer_id`, and `artist_id` columns with a data type
-of "integer number", a `billing_city` column with a data type of "text", a
-number of other columns, and and a `total` column that contains floating-point
-numbers.
+A relation is composed of rows. Each row in a relation contains a value for each
+of the relation's columns. Each column in a relation has an unique name and a
+designated data type. The table above has columns named `invoice_id`and
```
Comment (max-sixty) [inline web/book/src/tutorial/relations.md]:
```suggestion
designated data type. The table above is a relation, and has columns named  `invoice_id`and
```

Comment (max-sixty) [reply web/book/src/tutorial/relations.md]: WDYT? I'm trying to find a way of making "relation" fit into this...

**Comment by max-sixty** [comment]
I'll merge because it's great and we can iterate. 

I took the liberty of adding one phrase — but totally fine to undo if you don't think it fits (hope that's OK).

File: web/book/src/tutorial/annotated_example.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# Annotated example
```
Comment (max-sixty) [inline web/book/src/tutorial/annotated_example.md:1]: We could unify this example with the example in the readme, and move it to the front page of the book ( #3173)

File: web/book/src/tutorial/aggregation.md (hunk 1 above)
Comment (richb-hanover) [reply web/book/src/tutorial/aggregation.md:12]: I prefer to encourage readers to use the Playground. It has the double benefit of showing the SQL, and also showing the result of the query with one easy copy/paste.

**Pull request merged**